docs/griptape-framework/misc/src/events_chunk_stream.py

```diff
@@ -0,0 +1,37 @@
+from griptape.drivers import OpenAiChatPromptDriver
+from griptape.events import ActionChunkEvent, EventBus, EventListener, TextChunkEvent
+from griptape.structures import Pipeline
+from griptape.tasks import ToolkitTask
+from griptape.tools import PromptSummaryTool, WebScraperTool
+
+
+def action_chunk_listener(event: ActionChunkEvent) -> None:
+    if event.tag is not None and event.name is not None and event.path is not None:
+        print(f"{event.name}.{event.tag} ({event.path}) ", end="", flush=True)
+    if event.partial_input is not None:
+        print(event.partial_input, end="", flush=True)
+
+
+EventBus.add_event_listeners(
+    [
+        EventListener(
+            lambda e: print(e.token, end="", flush=True),
+            event_types=[TextChunkEvent],
+        ),
+        EventListener(
+            action_chunk_listener,
+            event_types=[ActionChunkEvent],
+        ),
+    ]
+)
```

Comment:
Can this be simplified now that we have `__str__` on the events?

Reply:
`events_3.py` is the simplified version of this. this is an example showing how to listen on the sub event types

Reply:
We can still show listening on sub-types while using the `str` method right?

Reply:
tis possible